x-pack/plugins/alerting/server/task_runner/create_execution_handler.test.ts

```diff
@@ -525,9 +515,88 @@ describe('Create Execution Handler', () => {
       alertExecutionStore,
     });
 
-    expect(alertExecutionStore.numberOfTriggeredActions).toBe(2);
-    expect(alertExecutionStore.numberOfScheduledActions).toBe(3);
-    expect(alertExecutionStore.triggeredActionsStatus).toBe(ActionsCompletion.PARTIAL);
+    expect(alertExecutionStore.getNumberOfTriggeredActions()).toBe(2);
+    expect(alertExecutionStore.getTriggeredActionsStatus()).toBe(ActionsCompletion.PARTIAL);
+    expect(createExecutionHandlerParams.logger.debug).toHaveBeenCalledTimes(1);
     expect(actionsClient.enqueueExecution).toHaveBeenCalledTimes(2);
   });
+
+  test('Skips triggering actions for a specific action type when it reaches the limit for that specific action type', async () => {
+    const executionHandler = createExecutionHandler({
+      ...createExecutionHandlerParams,
+      actionsConfigMap: {
+        default: {
+          max: 4,
+        },
+        'test-action-type-id': {
+          max: 1,
+        },
+      },
+      actions: [
+        ...createExecutionHandlerParams.actions,
+        {
+          id: '2',
+          group: 'default',
+          actionTypeId: 'test-action-type-id',
+          params: {
+            foo: true,
+            contextVal: 'My other {{context.value}} goes here',
+            stateVal: 'My other {{state.value}} goes here',
+          },
+        },
+        {
+          id: '3',
+          group: 'default',
+          actionTypeId: 'test-action-type-id',
+          params: {
+            foo: true,
+            contextVal: '{{context.value}} goes here',
+            stateVal: '{{state.value}} goes here',
+          },
+        },
+        {
+          id: '4',
+          group: 'default',
+          actionTypeId: 'another-action-type-id',
+          params: {
+            foo: true,
+            contextVal: '{{context.value}} goes here',
+            stateVal: '{{state.value}} goes here',
+          },
+        },
+        {
+          id: '5',
+          group: 'default',
+          actionTypeId: 'another-action-type-id',
+          params: {
+            foo: true,
+            contextVal: '{{context.value}} goes here',
+            stateVal: '{{state.value}} goes here',
+          },
+        },
+      ],
+    });
+
+    alertExecutionStore = new AlertExecutionStore();
+
+    await executionHandler({
+      actionGroup: 'default',
+      context: {},
+      state: { value: 'state-val' },
+      alertId: '2',
+      alertExecutionStore,
+    });
+
+    expect(alertExecutionStore.getNumberOfTriggeredActions()).toBe(4);
```

Comment:
Can we add some tests for `getNumberOfScheduledActions()` here as well - or maybe that's going to be difficult to arrange?